internal/server/v2/observation/contained_in.go

```diff
@@ -120,6 +122,15 @@ func FetchContainedIn(
 				sort.Sort(ranking.CohortByRank(cohorts))
 				for _, cohort := range cohorts {
 					facet := util.GetFacet(cohort)
+					if filter != nil && filter.Domain != "" {
+						url, err := url.Parse(facet.ProvenanceUrl)
+						if err != nil {
+							return nil, err
+						}
+						if !strings.HasSuffix(url.Hostname(), filter.Domain) {
```

Comment:
Add a line of comment for the matching logic?

Reply:
Done